src/sql/src/plan/statement/ddl.rs

```diff
@@ -1083,6 +940,182 @@ pub fn plan_create_sources(
+                            schema_registry_config: Some(ccsr_config),
+                            confluent_wire_format: true,
+                        }
+                    } else {
+                        unreachable!("CSR seed resolution should already have been called")
+                    }
+                }
+            };
+
+            // TODO(bwm): move this into purify_format
+            if let Some(key_schema) = key_schema {
+                return Ok(SourceDataEncoding::KeyValue {
+                    key: DataEncoding::Avro(AvroEncoding {
+                        schema: key_schema,
+                        schema_registry_config: schema_registry_config.clone(),
+                        confluent_wire_format,
+                    }),
+                    value: DataEncoding::Avro(AvroEncoding {
+                        schema: value_schema,
+                        schema_registry_config,
+                        confluent_wire_format,
+                    }),
+                });
+            } else {
+                DataEncoding::Avro(AvroEncoding {
+                    schema: value_schema,
+                    schema_registry_config,
+                    confluent_wire_format,
+                })
+            }
+        }
+        Format::Protobuf {
+            message_name,
+            schema,
+        } => {
+            let descriptors = match schema {
+                sql_parser::ast::Schema::Inline(bytes) => strconv::parse_bytes(&bytes)?,
+                sql_parser::ast::Schema::File(_) => {
+                    unreachable!("File schema should already have been inlined")
+                }
+            };
+
+            DataEncoding::Protobuf(ProtobufEncoding {
+                descriptors,
+                message_name: message_name.to_owned(),
+            })
+        }
+        Format::Regex(regex) => {
+            let regex = Regex::new(regex)?;
+            DataEncoding::Regex(RegexEncoding { regex })
+        }
+        Format::Csv {
+            header_row,
+            n_cols,
+            delimiter,
+        } => {
+            let n_cols = if col_names.is_empty() {
+                match n_cols {
+                    Some(n) => *n,
+                    None => bail!(
+                        "Cannot determine number of columns in CSV source; specify using \
+                             CREATE SOURCE...FORMAT CSV WITH X COLUMNS"
+                    ),
+                }
+            } else {
+                col_names.len()
+            };
+            DataEncoding::Csv(CsvEncoding {
+                header_row: *header_row,
+                n_cols,
+                delimiter: match *delimiter as u32 {
+                    0..=127 => *delimiter as u8,
+                    _ => bail!("CSV delimiter must be an ASCII character"),
+                },
+            })
+        }
+        Format::Json => unsupported!("JSON sources"),
+        Format::Text => DataEncoding::Text,
+    }))
+}
```

Comment:
I can see why you wanted to split these functions out but it's real hard to see what changed as a result :/

Reply:
I'll revert this.

Reply:
was this addressed?

Reply:
whoops, no, it is now though.